app/bundles/EmailBundle/Swiftmailer/Momentum/Facade/MomentumFacade.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace Mautic\EmailBundle\Swiftmailer\Momentum\Facade;
+
+use Mautic\EmailBundle\Swiftmailer\Momentum\Adapter\AdapterInterface;
+use Mautic\EmailBundle\Swiftmailer\Momentum\Callback\MomentumCallback;
+use Mautic\EmailBundle\Swiftmailer\Momentum\Exception\Facade\MomentumSendException;
+use Mautic\EmailBundle\Swiftmailer\Momentum\Service\SwiftMessageServiceInterface;
+use Mautic\EmailBundle\Swiftmailer\Momentum\Validator\SwiftMessageValidator\SwiftMessageValidatorInterface;
+use Monolog\Logger;
+
+/**
+ * Class MomentumApiFacade.
+ */
+final class MomentumFacade implements MomentumFacadeInterface
```

Comment:
I am not sure with this class - It holds a lot of logic. I think it would be better to split it to separate classes to make it easier to test.

Reply:
I found only handling response logic and nothing else. That can be moved into call of processImmediateFeedback and condition there. 

Rest of the code is just facade calling other logic in determined order. But otherwise pretty dumb.